git-repository/src/repository/state.rs

```diff
@@ -0,0 +1,42 @@
+use crate::RepositoryState;
+
+impl crate::Repository {
+    /// Returns the status of an in progress operation on a repository or [`None`]
+    /// if nothing is happening.
+    pub fn in_progress_operation(&self) -> Option<RepositoryState> {
```

Comment:
I experimented with this and noticed that `RepositoryState` has a `RepositoryState::None` item, but this function uses `Option::None` instead.